src/lib/validation.ts

```diff
@@ -55,32 +57,48 @@ export function checkString(s: unknown): string | void {
 	return s;
 }
 
+/**
+ * checkBooleanish
+ *
+ * Given an unknown value, determine if it can be evaluated to truthy/falsey.
+ *
+ */
+export function checkBooleanish(v: unknown): boolean {
+	return checkTruthy(v) || checkFalsey(v);
+}
+
 /**
  * checkTruthy
  *
- * Given a value which can be a string, boolean or number, return a boolean
- * which represents if the input was truthy
+ * Given an unknown value, determine if it evaluates to true.
+ *
  */
-export function checkTruthy(v: unknown): boolean | undefined {
+export function checkTruthy(v: unknown): boolean {
 	if (_.isString(v)) {
 		v = v.toLowerCase();
 	}
-	switch (v) {
-		case '1':
-		case 'true':
-		case true:
-		case 'on':
-		case 1:
-			return true;
-		case '0':
-		case 'false':
-		case false:
-		case 'off':
-		case 0:
-			return false;
-		default:
-			return;
+	return (
+		TRUTHY.filter((t) => {
+			return t === v;
+		}).length > 0
+	);
+}
+
+/**
+ * checkFalsey
+ *
+ * Given an unknown value, determine if it evaluates to false.
+ *
+ */
+export function checkFalsey(v: unknown): boolean {
+	if (_.isString(v)) {
+		v = v.toLowerCase();
 	}
+	return (
+		FALSEY.filter((f) => {
+			return f === v;
+		}).length > 0
+	);
```

Comment:
```suggestion
	return FALSEY.includes(v);
```

Reply:
I agree but not sure how to work around the following issue when using includes.

> Argument of type 'unknown' is not assignable to parameter of type 'string | number | boolean'.
  Type 'unknown' is not assignable to type 'true'.ts(2345)

I avoid the use of as any but what do you think about `return TRUTHY.includes(v as any)` if we really wanted to avoid the filter.

Reply:
I saw a good amount of instances use `as any` so I changed my mind.

Reply:
In this specific case of validation I don't mind the cast, although you could also do it via
```ts
const FALSEY: unknown[] = ['0', 'false', false, 'off', 0];